addons/account/models/account_move_line.py

```diff
@@ -2505,9 +2505,15 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
         expense_exchange_account = company.expense_currency_exchange_account_id
         income_exchange_account = company.income_currency_exchange_account_id
 
+        temp_exchange_move = self.env['account.move'].new({'journal_id': journal.id})
+        accounting_exchange_date = temp_exchange_move._get_accounting_date(
+            exchange_date or fields.Date.today(),
```

Comment:
Always use `context_today`:
```suggestion
            exchange_date or fields.Date.context_today(self),
```